gcp_pilot/healthcare.py

```diff
@@ -519,3 +519,35 @@ def get_resource_history(
 
         for entry in entries:
             yield resource.__class__(**entry["resource"])
+
+    def validate_resource(
+        self,
+        resource: DomainResource,
+        store_name: str,
+        dataset_name: str,
+        project_id: str = None,
+        location: str = None,
+        profile: str = None,
+    ):
+        parent = self._store_path(
+            name=store_name,
+            dataset_name=dataset_name,
+            project_id=project_id,
+            location=location,
+        )
+        url = f"{self._base_url}/{parent}/fhir/{resource.get_resource_type()}/$validate"
+
+        if profile:
+            url = f"{url}?profile={profile}"
+
+        params = {
+            "url": url,
+            "headers": {"Content-Type": "application/fhir+json;charset=utf-8"},
+            "json": as_json(resource),
+        }
+
+        try:
+            self._execute(method=self._session.post, **params)
```

Comment:
We should try to avoid using plain URLs as much as possible because gcp library already has a builtin URL discovery system (it's very very weird, but it's way safer than string interpolation)
I know that even the documentation has an example using the URL, but this only happens to very early-stage APIs that they did not have much time properly documenting it.

Here's an example os the same call using the endpoint discovery:
```
     parent = self._store_path(
            name=store_name,
            dataset_name=dataset_name,
            project_id=project_id,
            location=location,
        )

      try:
          self._execute(
              method=self.client.projects().locations().datasets().fhirStores().fhir().Resource_validate,
              parent=parent,
              type=resource.get_resource_type(),
              body=as_json(resource),
          )
```